designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/scenariostatus/ScenarioStatusProvider.scala

```diff
@@ -78,6 +80,36 @@ class ScenarioStatusProvider(
     )
   }
 
+  def getActiveScenariosFor(
+      processingTypes: Iterable[ProcessingType]
+  )(implicit user: LoggedUser): IO[Set[ProcessName]] = {
+    IO.fromFuture {
+      IO {
+        implicit val freshnessStatus: DataFreshnessPolicy = DataFreshnessPolicy.Fresh
+        dbioRunner.run(for {
+          scenarios <- processRepository.fetchLatestProcessesDetails[Unit](
+            ScenarioQuery.empty
+              .copy(isFragment = Some(false), isArchived = Some(false), processingTypes = Some(processingTypes))
+          )
+          activeScenarios <- DBIOAction
+            .sequence {
+              scenarios.map { process =>
+                for {
+                  inProgressActionNames <- actionRepository.getInProgressActionNames(process.processId)
+                  scenarioStatus <- getScenarioStatusFetchingDeploymentsStatusesFromManager(
```

Comment:
There is another corner case here. At the end of this method is invoked `InconsistentStateDetector`. It can return `ProblemStateStatus.shouldNotBeRunning(true)` which means that scenario s running but based on the local, designer state it is an unexpected situation. Do you think that we should care about? Currently, you count it as inactive scenario but in fact it runs on Flink.

Reply:
Yes, but I assumed that users should be able to do sth with the limitation. 
I mean, if there is e.g., a restarting scenario (which exceeds the limit), they can cancel it and run the other scenario. 

In the use case you described, a user has no way to fix the problem. AFAIU, we detect it, but there is no exact cause for it. It may happen because of reasons. So, IMO, we should not "punish" users for that. 

That's why I'd not take it into consideration in case of limits.

Reply:
In case of this problem, I think user can still cancel this scenario, the same as with restarting scenario. They also can redeploy it to show to designer that on Flink is the correct state. It is why in `ProblemStateStatus.defaultActions` are both `deploy` and `cancel` actions.

Reply:
ok, it makes sense if users can cancel the scenario in the problem status. I modified the code to handle this case too.